src/Auth/BaseAuthenticate.php

```diff
@@ -139,7 +142,7 @@ protected function _findUser($username, $password = null)
      * @param string $username The username/identifier.
      * @return \Cake\ORM\Query
      */
-    protected function _query($username)
+    protected function _query(string $username): Query
```

Comment:
If you force-return something here, it sure shouldnt be a Query object, but a true contract (interface) or alike, the concrete ORM level object on the Auth namespace doesn't sound like a good idea for such a hard return type fixation.

Reply:
So the annotation is simply wrong here?

Reply:
If every typehint needs to be an interface we won't have many typehints to add. Adding interfaces for each class so it can be used in a typehint seems tedious and against the goal of helping ides and preventing accidents.

Reply:
Adding interfaces for each type of class is neither practical or necessary.

Reply:
The added `use Cake\ORM\Query;` statement shows that up until now there was no hard coupling between two very concrete objects. That now changes.
But I can see how the framework is yet still very far away of doing interface contracting instead of hard class object coupling across different layers etc.
So never mind me here :)

Reply:
Skipping the typehints and continuing to rely on documentation annotations and phpstan is an option. I think that unless we have a guideline on what kinds of classes can be used in typehints we're going to spend a lot of time discussing these matters.

Reply:
We can also fix up return types to a better/cleaner approach in 5.0.
So I guess this is OK here for now.